qemu/qmp.go

```diff
@@ -1136,3 +1176,24 @@ func (q *QMP) ExecuteVirtSerialPortAdd(ctx context.Context, id, name, chardev st
 
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
+
+// ExecuteQueryMigration queries migration progress.
+func (q *QMP) ExecuteQueryMigration(ctx context.Context) (MigrationStatus, error) {
+	var status MigrationStatus
+
+	response, err := q.executeCommandWithResponse(ctx, "query-migrate", nil, nil, nil)
+	if err != nil {
+		return status, err
+	}
+
+	data, err := json.Marshal(response)
+	if err != nil {
+		return status, fmt.Errorf("Unable to extract migrate status information: %v", err)
+	}
+
+	if err = json.Unmarshal(data, &status); err != nil {
+		return status, fmt.Errorf("Unable to convert migrate status information: %v", err)
```

Comment:
I think this could return a partially constructed status if we get an error.  It might be better to return MigrationStatus{} just in case.